docs/architecture/adr-004-genesis-modes.md

```diff
@@ -0,0 +1,186 @@
+# ADR 4: Multiple Genesis modes
+
+## Changelog
+
+- July 26, 2019: Initial draft
+
+## Context
+
+### Problem
+
+Currently, some of the modules `GenesisState` are populated only if the corresponding
+object passed through the Genesis JSON is empty (_i.e_ as an empty array) or if it's
+undefined or the value is zero.
+
+This leads to confusion as you might pass a valid genesis file which contains zero/empty
+values and have `InitGenenesis` perform this unnecessary check.
+
+### Proposed Solution
```

Comment:
Perhaps it's best to join the "Proposal" with what's outlined in the "Decision".